pennylane_lightning/lightning_tensor/_measurements.py

```diff
@@ -207,9 +219,167 @@ def measure_tensor_network(self, circuit: QuantumScript) -> Result:
         """
 
         if circuit.shots:
-            raise NotImplementedError("Shots are not supported for tensor network simulations.")
+            # finite-shot case
+            results = self.measure_with_samples(
+                circuit.measurements,
+                shots=circuit.shots,
+            )
+
+            if len(circuit.measurements) == 1:
+                if circuit.shots.has_partitioned_shots:
+                    return tuple(res[0] for res in results)
+
+                return results[0]
+
+            return results
         # analytic case
         if len(circuit.measurements) == 1:
             return self.measurement(circuit.measurements[0])
 
         return tuple(self.measurement(mp) for mp in circuit.measurements)
+
+    # pylint:disable = too-many-arguments
+    def measure_with_samples(
```

Comment:
We should be able to use these methods from `_measurements_base` at some point in a follow up PR.